module/VuFind/src/VuFind/AjaxHandler/GetSearchResults.php

```diff
@@ -0,0 +1,455 @@
+        SessionSettings $sessionSettings,
+        ResultsManager $resultsManager,
+        PhpRenderer $renderer,
+        RecordLoader $recordLoader,
+        ?UserRow $user,
+        string $sessionId,
+        SearchNormalizer $normalizer,
+        Search $searchTable,
+        array $config,
+        VersionsHelper $versionsHelper
+    ) {
+        $this->sessionSettings = $sessionSettings;
+        $this->resultsManager = $resultsManager;
+        $this->renderer = $renderer;
+        $this->recordLoader = $recordLoader;
+        $this->user = $user;
+        $this->sessionId = $sessionId;
+        $this->searchNormalizer = $normalizer;
+        $this->searchTable = $searchTable;
+        $this->config = $config;
+        $this->versionsHelper = $versionsHelper;
+    }
+
+    /**
+     * Handle a request.
+     *
+     * @param Params $params Parameter helper from controller
+     *
+     * @return array [response data, HTTP status code]
+     */
+    public function handleRequest(Params $params)
+    {
+        $results = $this->getSearchResults($params);
+        if (!$results) {
+            return $this->formatResponse(['error' => 'Invalid request'], 400);
+        }
+        $elements = $this->getElements($params, $results);
+        return $this->formatResponse(compact('elements'));
+    }
+
+    /**
+     * Get search results
+     *
+     * @param Params $params Request params
+     *
+     * @return ?Results
+     */
+    protected function getSearchResults(Params $params): ?Results
+    {
+        parse_str($params->fromQuery('querystring', ''), $searchParams);
+        $backend = $params->fromQuery('source', DEFAULT_SEARCH_BACKEND);
+        $searchType = $params->fromQuery('searchType', '');
+        if ('versions' === $searchType) {
+            $driverAndKeys = $this->versionsHelper->getDriverAndWorkKeysFromParams(
+                $searchParams,
+                $backend
+            );
+            if (empty($driverAndKeys['keys'])) {
+                return null;
+            }
+
+            $searchParams['lookfor'] = $this->versionsHelper
+                ->getSearchStringFromWorkKeys((array)$driverAndKeys['keys']);
+            $searchParams['type'] = $this->versionsHelper->getWorkKeysSearchType();
+        }
+
+        $results = $this->resultsManager->get($backend);
+        $paramsObj = $results->getParams();
+        $paramsObj->getOptions()->spellcheckEnabled(false);
+        $paramsObj->initFromRequest(new Parameters($searchParams));
+
+        if ('versions' !== $searchType) {
+            $this->saveSearchToHistory($results);
+        }
+
+        return $results;
+    }
+
+    /**
+     * Render page elements
+     *
+     * @param Params  $params  Request params
+     * @param Results $results Search results
+     *
+     * @return array
+     */
+    protected function getElements(Params $params, Results $results): array
+    {
+        if ($params->fromQuery('searchType', '') === 'versions') {
```

Comment:
`$this->customizeUrlHelperForSearchType($results, $params)` for abstraction?

Reply:
No longer needed.